File: level_1/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = []
    stack = []
    count = 0
    for i in moves:
        for j in range(len(board)):
            if board[j][i-1] > 0:
                stack.append(board[j][i-1])
                board[j][i-1] = 0
                if stack[-1:] == stack[-2:-1]:
                    #배열의 마지막 요소를 넣는다
                    answer += stack[-1:]
                    #중복된 두개를 pop
                    stack = stack[:-2]
                #다음을 탐색한다    
                break
                
    count = len(answer) * 2
    return count

File: 코딩테스트_합격자되기_자바스크립트/스택/크레인 인형 뽑기 게임.js
function solution(board, moves) {
  // 2차원 배열에서 가장 상단에 있는 내역을 알 수 있는 방법이 중요하겄는데
  // n * n을 순회 해야함

  const stack = [];
  let result = 0;

  for (const move of moves) {
    for (const arr of board) {
      // 만약 0이 아닌 값이 있다면
      if(arr[move - 1]) {
        // stack 자료구조 사용
        if(arr[move - 1] === stack[stack.length - 1]) {
          // top이랑 같은 값이라면 pop후 result ++;
          stack.pop();
          result++;
        } else {
          stack.push(arr[move - 1]);
        }
        // 뽑았으면 0으로 변환
        arr[move - 1] = 0;
        break;
      }
    }
  }
  return result * 2;
}